packages/kbn-esql-validation-autocomplete/src/autocomplete/__tests__/autocomplete.command.stats.test.ts

```diff
@@ -285,6 +290,33 @@ describe('autocomplete.suggest', () => {
           [',', '| ', '+ $0', '- $0']
         );
       });
+      test('on space within bucket()', async () => {
+        const { assertSuggestions } = await setup();
+        await assertSuggestions('from a | stats avg(b) by BUCKET(/, 50, ?t_start, ?t_end)', [
+          // Note there's no space or comma in the suggested field names
+          ...getFieldNamesByType(['date', ...ESQL_COMMON_NUMERIC_TYPES]),
+          ...getFunctionSignaturesByReturnType('eval', ['date', ...ESQL_COMMON_NUMERIC_TYPES], {
+            scalar: true,
+          }),
+        ]);
+        await assertSuggestions('from a | stats avg(b) by BUCKET(  /  , 50, ?t_start, ?t_end)', [
+          // Note there's no space or comma in the suggested field names
+          ...getFieldNamesByType(['date', ...ESQL_COMMON_NUMERIC_TYPES]),
+          ...getFunctionSignaturesByReturnType('eval', ['date', ...ESQL_COMMON_NUMERIC_TYPES], {
+            scalar: true,
+          }),
+        ]);
```

Comment:
Thanks for adding these tests. Another option would be to use `suggest` directly and then check explicitly that none of the suggestions include a trailing space. This might clarify the point of the assertion.

```ts
const { suggest } = await setup();
const suggestions = await suggest('from a | stats avg(b) by BUCKET(/, 50, ?t_start, ?t_end)');
expect(!suggestions.some(s => s.text.endsWith(' ')));
```

Reply:
This is a great suggestion, although I would like to keep the tests as is as the tested cases are more strict. I realized we were missing this test cases anyway to check if the right things are being suggested where the cursor is supposed to be.